### PR TITLE
Refactor parameters and gamesession

### PR DESCRIPTION
**Pull request by 9gix**

### CODE REVIEW

**Comment by bloodelves88** [comment]
This error happens after launching:
```
D:\Work\Repositories\CloudGameThinClient-9gix\thin_client>python main.py 127.0.0.1 30000 0
Traceback (most recent call last):
  File "main.py", line 144, in <module>
    main(sys.argv[1], sys.argv[2], sys.argv[3])
  File "main.py", line 129, in main
    start_client(ip, port, int(player_controller_id))
  File "main.py", line 124, in start_client
    action.process(event)
  File "main.py", line 20, in process
    raise NotImplementedError
NotImplementedError
```
After I fix that, this is the next problem:
```
D:\Work\Repositories\CloudGameThinClient-9gix\thin_client>python main.py 127.0.0.1 30000 0
Traceback (most recent call last):
  File "main.py", line 144, in <module>
    main(sys.argv[1], sys.argv[2], sys.argv[3])
  File "main.py", line 129, in main
    start_client(ip, port, int(player_controller_id))
  File "main.py", line 124, in start_client
    action.process(event)
  File "main.py", line 46, in process
    self.session.pack_and_send(settings.DEVICE_MOUSE, x, y, event.type)
  File "D:\Work\Repositories\CloudGameThinClient-9gix\thin_client\session.py", line 35, in pack_and_send
    data_keyboard = (settings.VERSION, device_type, self.sequence, controller_id,
AttributeError: 'GameSession' object has no attribute 'sequence'
```

**Comment by 9gix** [comment]
@bloodelves88 what `event.type` is that such that it reach to the unimplemented `Action`?

**Comment by bloodelves88** [comment]
Seems to be some pygame startup/initialization events:
```
<Event(17-VideoExpose {})>
<Event(16-VideoResize {'size': (640, 480), 'h': 480, 'w': 640})>
<Event(1-ActiveEvent {'state': 1, 'gain': 0})>
<Event(4-MouseMotion {'rel': (640, 413), 'buttons': (0, 0, 0), 'pos': (639, 412)})>
<Event(1-ActiveEvent {'state': 1, 'gain': 1})>
```

File: thin_client/session.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                         socket.SOCK_DGRAM) # UDP
+        self.ip_address = ip_address
+        self.player_controller_id = player_controller_id
+        self.sequence = 0
+
+    def pack_and_send(self, device_type, ue_key_code, 
+                      ue_char_code, event_type):
+        data_keyboard = (settings.VERSION, device_type, self.sequence, controller_id,
+                         ue_key_code, ue_char_code, event_type)
+        data_mouse = (settings.VERSION, device_type, self.sequence, controller_id,
```
Comment (bloodelves88) [inline thin_client/session.py]: `controller_id` needs to be changed to `self.player_controller_id`

**Comment by bloodelves88** [comment]
I fix the `NotImplementedError` by commenting it out and adding `pass`

**Comment by 9gix** [comment]
While fixing the test, I notice that you have 3 functions just to test a sequence increment, it could be improved by removing the other 2 because they are redundant.

Last time, I remember I ask for separation, but that is only for the case if each test does different things.

File: thin_client/main.py [hunk 2]
```diff
@@ -29,35 +33,35 @@ def process(self, player_controller_id, sequence, sock, pygame, event, ip_addres
...
-                                          player_controller_id, ue_key_code,
-                                          ue_char_code, event.type, sock, ip_address)
+        self.session.pack_and_send(settings.DEVICE_KEYBOARD, 
+            ue_key_code, ue_char_code, event.type)
 
         logging.info("Keyboard: %s => %s", event.key, ue_key_code)
 
+class QuitAction(Action):
+    def process(self, event):
+        self.session.send_quit_command(session)
```
Comment (bloodelves88) [inline thin_client/main.py]: `send_quit_command` does not take in a session parameter

**Pull request merged**